Kernel/Net/Intel/E1000NetworkAdapter.cpp

```diff
@@ -159,6 +159,16 @@ UNMAP_AFTER_INIT static bool is_valid_device_id(u16 device_id)
     }
 }
 
+UNMAP_AFTER_INIT ErrorOr<NonnullLockRefPtr<E1000NetworkAdapter>> E1000NetworkAdapter::create(PCI::Address address, u8 irq, NonnullOwnPtr<IOWindow> registers_io_window, NonnullOwnPtr<KString> interface_name)
```

Comment:
This seems overly complicated and unnecessary. Why do you create a new method instead of just allocating these regions in the `try_to_initialize` method and then to pass it to the constructor?
Please change this to use the pattern we use in all other kernel drivers.

Reply:
Okay, will do. I originally did this because I thought there was a distinction between initializing the adapter and creating it, but it does make more sense to just follow the pattern used in other drivers :)

Reply:
This is not what I meant - it was meant that you will create the regions before constructing the device and then pass them to the constructor. I will see how to fix it.